### DISCUSSION TITLE
In house obfuscation for source code

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/roslyn#66193] [Ideas]
If roslyn can produce obfuscated codes it can make IP theft more difficult, therefore protects our business partners.
The people who are building roslyn should be able to do a better job than some half ass virtual machine solution out in the wild too

**Comment by CyrusNajmabadi** [comment]
Closing out. This is not a goal of roslyn.  There are already experts in this space with solutions available. 

> The people who are building roslyn should be able to do a better job 

Unlikely. This is not an area we are experts in. Other companies have a tremendous amount of knowledge and skill in this arena. You should go with them if this is important to you.
